topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlHandler.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.cruisecontrol;
+
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+import io.strimzi.api.kafka.model.topic.ReplicasChangeStatusBuilder;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.topic.TopicOperatorConfig;
+import io.strimzi.operator.topic.TopicOperatorUtil;
+import io.strimzi.operator.topic.cruisecontrol.CruiseControlClient.TaskState;
+import io.strimzi.operator.topic.metrics.TopicOperatorMetricsHolder;
+import io.strimzi.operator.topic.model.ReconcilableTopic;
+import io.strimzi.operator.topic.model.Results;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.strimzi.api.kafka.model.topic.ReplicasChangeState.ONGOING;
+import static io.strimzi.api.kafka.model.topic.ReplicasChangeState.PENDING;
+import static io.strimzi.operator.topic.TopicOperatorUtil.hasReplicasChangeStatus;
+import static io.strimzi.operator.topic.TopicOperatorUtil.topicNames;
+import static java.lang.String.format;
+import static org.apache.logging.log4j.core.util.Throwables.getRootCause;
+
+/**
+ * Handler for Cruise Control requests.
+ */
+public class CruiseControlHandler {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(CruiseControlHandler.class);
+    
+    private final TopicOperatorConfig config;
+    private final TopicOperatorMetricsHolder metricsHolder;
+    private final CruiseControlClient cruiseControlClient;
+
+    /**
+     * Create a new instance.
+     *
+     * @param config Topic Operator configuration.
+     * @param metricsHolder Metrics holder.
+     * @param cruiseControlClient Cruise Control client.
+     */
+    public CruiseControlHandler(TopicOperatorConfig config,
+                                TopicOperatorMetricsHolder metricsHolder,
+                                CruiseControlClient cruiseControlClient) {
+        this.config = config;
+        this.metricsHolder = metricsHolder;
+        this.cruiseControlClient = cruiseControlClient;
+    }
+
+    /**
+     * Send a topic_configuration request to create a task for replication factor change of one or more topics.
+     * This should be called when one or more .spec.replicas changes are detected.
+     * Note that this method also updates the KafkaTopic status.
```

Comment:
If we can't avoid performing this side effect let's at least make this `<strong>`

Reply:
I forgot to remove the note, these pending status updates are now also stored in Results.